cargo-screeps/README.md

```diff
@@ -46,7 +46,114 @@ This is not fully tested, but feel free to use! Issues are welcome.
   - runs `cargo web --check --target=wasm32-unknown-unknown` which is fairly similar to
     `cargo check`.
 
+# Configuration Options
+
+## No namespace
+
+- `default_deploy_mode`: what `cargo screeps deploy` does: use `"copy"` or `"upload"`
+
+## `[upload]`
+
+This configures options specific to the `cargo screeps upload` deploy mode.
+
+- `username`: your Screeps username or email
+- `password`: your Screeps password. For private servers set a password using [screepsmod-auth].
+- `branch`: the branch on the server to upload files to
+- `ptr`: if true, upload to the "ptr" realm
+- `hostname`: the hostname to upload to. For example, `screeps.com` or `localhost` or `server1.screepsplu.us`
+- `ssl`: whether to connect to the server using ssl. Should be false for private servers
+- `port`: port to connect to server with. Should generally be `21025` for private servers
+
+## `[copy]`
+
+This configures options specific to the `cargo screeps copy` deploy mode.
+
+- `destination`: the directory to copy files into. can be relative to `screeps.toml` or absolute
+- `branch`: the "branch" to copy into. This is a subdirectory of `destination` which the js/wasm files will be copied into.
+- `prune`: if true, any files in the destination directory which were not just copied will be deleted after copying.
+
+## `[build]`
+
+This configures general build options.
+
+- `output_js_file`: the javascript file to export bindings and bootstrapping as (default `"main.js"`)
+- `output_wasm_file`: the WASM file to rename compile WASM to (default `"compiled.wasm"`)
+- `initialize_header_file`: a file containing the JavaScript for starting the WASM instance. See
+  [overriding the default initialization header](#overriding-the-default-initialization-header)
+
+## Overriding the default initialization header
+
+`cargo-screeps` tries to make a reasonable `main.js` file to load the WASM. However, it's pretty
+basic, and you might find you want to do some things in JavaScript before loading the WASM module.
+
+Luckily, you can override this initialization! Set `build.initialize_header_file` to a file containing the JavaScript initialization code.
+
+The code will have access to two utility functions, `wasm_fetch_module_bytes` and `wasm_create_stdweb_vars`:
```

Comment:
Wouldn't that part be much easier to understand if you made a template or maybe just commented `default_initialization_header`?

Reply:
I'm not sure? I mean I wrote this section with the intention of someone who knows the WebAssembly JavaScript types. I thought that giving a clear idea of what someone writing a custom function has access to would be better than _only_ example code.

I can definitely make more documentation here and write more example code, better commented, but I think showing the functions one can call provides value that well-commented example code doesn't.

Reply:
Of course, I think I meant "in addition of".